packages/sdks/nextjs-sdk/rollup.config.mjs

```diff
@@ -65,45 +65,77 @@ const configurations = ['server', 'client', ''].map((entry) => {
 				return;
 			}
 			warn(warning);
+		}
+	};
+
+	return [
+		{
+			...baseConf,
+			output: {
+				dir: cjsOutputDir,
+				sourcemap: true,
+				format: 'cjs',
+				preserveModules: true,
+				exports: 'auto'
+			},
+			plugins: commonPlugins(cjsOutputDir)
 		},
-		// externals: ['./shared'],
-		output: [
-			// {
-			//   // file: outputPath,
-			// 	dir: outputDir,
-			//   sourcemap: true,
-			//   format: 'cjs',
-			// 	preserveModules: true
-			// },
-			{
-				// file: outputPath.replace('.js', '.mjs'),
-				dir: outputDir,
+		{
+			...baseConf,
+			output: {
+				dir: esmOutputDir,
 				sourcemap: true,
 				format: 'esm',
 				preserveModules: true
-			}
-		],
-		plugins: commonPlugins(outputDir)
-	};
+			},
+			plugins: commonPlugins(esmOutputDir)
+		}
+	];
 });
 
-const endConfigurations = ['server', 'client', ''].map((entry) => {
-	const input = entry
-		? `./dist/dts/src/${entry}/index.d.ts`
-		: './dist/dts/src/index.d.ts';
-	const outputFile = entry ? `dist/${entry}/index.d.ts` : 'dist/index.d.ts';
-	const srcDir = entry ? `./dist/${entry}/src` : './dist/src';
-	return {
-		input: input,
-		output: [{ file: outputFile, format: 'esm' }],
-		plugins: [
-			dts(),
-			del({
-				hook: 'buildEnd',
-				targets: srcDir
-			})
-		]
-	};
+const endConfigurations = ['server', 'client', ''].flatMap((entry) => {
```

Comment:
Do you think we can simplify the config and include it somehow in the config above?
Why do we need to split the configs and merge it later?